rosbag2_cpp/src/rosbag2_cpp/writers/sequential_writer.cpp

```diff
@@ -432,6 +436,26 @@ bool SequentialWriter::should_split_bagfile(
   return should_split;
 }
 
+bool SequentialWriter::message_within_accepted_time_range(
+  const std::chrono::time_point<std::chrono::high_resolution_clock> & current_time) const
+{
+  bool ret_value = true;
+  if ( (storage_options_.start_time_ns > 0) &&
+    current_time < std::chrono::time_point<std::chrono::high_resolution_clock>(
+      std::chrono::nanoseconds(storage_options_.start_time_ns)) )
+  {
+    ret_value = false;
+  } else {
+    if ( (storage_options_.end_time_ns > 0) &&
+      current_time > std::chrono::time_point<std::chrono::high_resolution_clock>(
+        std::chrono::nanoseconds(storage_options_.end_time_ns)) )
+    {
+      ret_value = false;
+    }
+  }
+  return ret_value;
+}
```

Comment:
We can avoid at least 3 type conversions to the `std::chrono::time_point<std::chrono::high_resolution_clock>(timestamp)` if we would be using  `uint32_t` type for the `start_time_ns` and `end_time_ns` parameters and use native comparison for timestamps. It will be better from the performance point of view.

Reply:
Or maybe using `int64_t` type in this case 0 will be a valid range and we can default to the -1 to determine that start or end time limit is not set.

Reply:
The type conversions should be optimised away by the compiler (I haven't tested it but could look into it if you think it is important)?

Are we optimising for 64bit or 32bit machines - the 2^32 nanoseconds are only ~4 seconds so 64bit is needed I think?

Reply:
I have now simplified message_within_accepted_time_range() and now it uses int64_t.

I did look into performance also. Converting a 900MB mcap takes ~25s on my Mac M2.
The simplified code did not change the performance noticeably.
As a matter of fact - the writing of the new .mcap (and the within-check) is negligible. Writing to 2 new files simultaneously takes also ~25s. I guess the grunt work is happening in the reading.